src/widgets/menu/SideMenuItem.tsx

```diff
@@ -1,37 +1,77 @@
 import WidgetBase from '@dojo/framework/widget-core/WidgetBase';
 import { tsx } from '@dojo/framework/widget-core/tsx';
-import ThemedMixin, { theme } from '@dojo/framework/widget-core/mixins/Themed';
-import I18nMixin from '@dojo/framework/widget-core/mixins/I18n';
+import ThemedMixin, { theme, ThemedProperties } from '@dojo/framework/widget-core/mixins/Themed';
 import ActiveLink from '@dojo/framework/routing/ActiveLink';
 import { Params } from '@dojo/framework/routing/interfaces';
 
+import FontAwesomeIcon from '../icon/FontAwesomeIcon';
+
 import * as css from './SideMenuItem.m.css';
 
-export interface SideMenuItemProperties {
-	to: string;
+export interface SideMenuItemProperties extends ThemedProperties {
+	name?: string;
+	to?: string;
 	params?: Params;
+	inverse?: boolean;
 }
 
 @theme(css)
-export default class SideMenuItem extends ThemedMixin(I18nMixin(WidgetBase))<SideMenuItemProperties> {
+export default class SideMenuItem extends ThemedMixin(WidgetBase)<SideMenuItemProperties> {
+	private _dropDownOpen = false;
+	private _toggleDropDown() {
+		this._dropDownOpen = !this._dropDownOpen;
+		this.invalidate();
+	}
+
+	private _renderDropDownLink() {
+		const { name = '', inverse = false } = this.properties;
+
+		const url = window.location.pathname;
+		let activeClass: string | undefined = undefined;
+		if (name !== '' && new RegExp(name.toLowerCase().replace(' ', '-'), 'g').test(url)) {
```

Comment:
This would be more efficient as a `.indexOf` or (more modern) `.includes` since it doesn't force a regex engine compile.